app/services/SortStatementsService.scala

```diff
@@ -27,36 +28,49 @@ class SortStatementsService @Inject()() {
 
   def sortDutyDefermentStatementsForEori(historicEori: EoriHistory,
                                          dutyDefermentFiles: Seq[DutyDefermentStatementFile]): DutyDefermentStatementsForEori = {
-    dutyDefermentFiles.partition(_.metadata.statementRequestId.isEmpty) match {
-      case (current, requested) => DutyDefermentStatementsForEori(historicEori, current, requested)
+    dutyDefermentFiles.partition(_.metadata.statementRequestId.isDefined) match {
+      case (requested, current) => DutyDefermentStatementsForEori(historicEori, current, requested)
```

Comment:
Are we intending to swap the "Requested, Current" around?

Reply:
@AdamCoops Identified an edge case in prod where a statement group can have both regular and requested statements. To fix the issue, swapped the current and requested around. Also, partition logic is modified to check for isDefined instead of isEmpty